proxy/src/timeout.rs

```diff
@@ -212,3 +219,42 @@ where
            .finish()
     }
 }
+
+//===== impl HumanDuration =====
+
+impl fmt::Display for HumanDuration {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        let secs = self.as_secs();
+        let subsec_ms = self.subsec_nanos() as f64 / 1_000_000f64;
+        if secs == 0 {
+            write!(fmt, "{} ms", subsec_ms)
```

Comment:
tioli: i'd prefer to omit the space between the value and unit so log lines end up as e.g.

> `ERROR 2018-04-04T20:05:49Z: conduit_proxy: turning operation timed out after 100ms into 500`